x-pack/plugins/lens/public/datasources/form_based/form_based.tsx

```diff
@@ -219,6 +219,14 @@ export function getFormBasedDatasource({
       const newLayers = { ...state.layers };
       delete newLayers[layerId];
 
+      // delete layers linked to this layer
+      Object.keys(newLayers).forEach((id) => {
+        const linkedLayers = newLayers[id]?.linkToLayers;
+        if (linkedLayers && linkedLayers.length && linkedLayers.includes(layerId)) {
```

Comment:
nit: I think one check can be removed:
```suggestion
        if (linkedLayers && linkedLayers.includes(layerId)) {
```